api/envoy/config/core/v3/substitution_format_string.proto

```diff
@@ -113,4 +119,7 @@ message SubstitutionFormatString {
   // See the formatters extensions documentation for details.
   // [#extension-category: envoy.formatter]
   repeated TypedExtensionConfig formatters = 6;
+
+  // If json_format is used, the options will be applied to the output JSON string.
+  JsonFormatOptions json_format_options = 7;
```

Comment:
I think ``text_format`` is a bit different because wraps multiple source options rather than source + print option. Although generally your suggestion is possible I think deprecation does not worth it, but don't have a strong opinion.

Reply:
Yeah, considering the deprecation, I am also not sure if it's a good idea.

Anyway, this is a frequently-used API, it worth more consideration and discussions.

Reply:
yeah I agree in a perfect world we do the deprecation, but I don't think it's worth it either.